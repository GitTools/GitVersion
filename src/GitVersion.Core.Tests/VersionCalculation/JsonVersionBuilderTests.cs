using GitVersion.Core.Tests.Helpers;
using GitVersion.VersionCalculation;
using Microsoft.Extensions.DependencyInjection;
using NUnit.Framework;
using Shouldly;

namespace GitVersion.Core.Tests;

[TestFixture]
public class JsonVersionBuilderTests : TestBase
{
    [SetUp]
    public void Setup() => ShouldlyConfiguration.ShouldMatchApprovedDefaults.LocateTestMethodUsingAttribute<TestAttribute>();

    [Test]
    public void Json()
    {
        var semanticVersion = new SemanticVersion
        {
            Major = 1,
            Minor = 2,
            Patch = 0,
            PreReleaseTag = "unstable4",
            BuildMetaData = new SemanticVersionBuildMetaData("versionSourceSha", 5, "feature1", "commitSha", "commitShortSha", DateTimeOffset.Parse("2014-03-06 23:59:59Z"), 0)
        };

        var configuration = new TestEffectiveConfiguration();

        var serviceProvider = ConfigureServices();

        var variableProvider = serviceProvider.GetRequiredService<IVariableProvider>();
        var variables = variableProvider.GetVariablesFor(semanticVersion, configuration, false);
        var json = variables.ToString();
        json.ShouldMatchApproved(c => c.SubFolder("Approved"));
    }
}
