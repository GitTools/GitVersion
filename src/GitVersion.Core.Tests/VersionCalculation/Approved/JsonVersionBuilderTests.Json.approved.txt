{
  "Major": 1,
  "Minor": 2,
  "Patch": 0,
  "PreReleaseTag": "unstable.4",
  "PreReleaseTagWithDash": "-unstable.4",
  "PreReleaseLabel": "unstable",
  "PreReleaseLabelWithDash": "-unstable",
  "PreReleaseNumber": 4,
  "WeightedPreReleaseNumber": 4,
  "BuildMetaData": 5,
  "FullBuildMetaData": "5.Branch.feature1.Sha.commitSha",
  "MajorMinorPatch": "1.2.0",
  "SemVer": "1.2.0-unstable.4",
  "AssemblySemVer": "1.2.0.0",
  "AssemblySemFileVer": "1.2.0.0",
  "FullSemVer": "1.2.0-unstable.4+5",
  "InformationalVersion": "1.2.0-unstable.4+5.Branch.feature1.Sha.commitSha",
  "BranchName": "feature1",
  "EscapedBranchName": "feature1",
  "Sha": "commitSha",
  "ShortSha": "commitShortSha",
  "VersionSourceSha": "versionSourceSha",
  "CommitsSinceVersionSource": 5,
  "UncommittedChanges": 0,
  "CommitDate": "2014-03-06"
}